### ISSUE TITLE
Allow public fields to be set by OptionParser

### ISSUE THREAD

**Issue by Insomniak47** [slagroom/Parsimony#9]
Currently because public fields are member access but aren't properties when they are turned into PropertySelectors they will throw an exception even though it's a valid thing to want to do. We should:

- [ ] Create a FieldAccess type that handles access to fields
- [ ] Add code to determine the nature of the expression type and construct the correct one
- [ ] Plumb them into the different places that need it

**Comment by ghost** [comment]
alternatively, the property expression type could handle both. a C# field isn't a C# property, but an object property is logically an object field. I have no strong preference, but since it's already exposing Func/Action and those are going to be needed to read/write the field it seems like it would make abstract layer above cleaner if we just pretended it's the same thing

**Comment by Insomniak47** [comment]
Yeah possibly, I don't think it ends up as a ton of work either way so I'll probably just poke at it over time. This was more for tracking
